app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/MockNavigationOffboardRouterActivity.kt

```diff
@@ -0,0 +1,372 @@
+import com.mapbox.navigation.base.route.DirectionsSession
+import com.mapbox.navigation.base.route.model.Route
+import com.mapbox.navigation.directions.session.MapboxDirectionsSession
+import com.mapbox.navigation.route.offboard.MapboxOffboardRouter
+import com.mapbox.navigation.utils.extensions.ifNonNull
+import com.mapbox.services.android.navigation.testapp.R
+import com.mapbox.services.android.navigation.testapp.activity.notification.CustomNavigationNotification
+import com.mapbox.services.android.navigation.testapp.utils.Utils
+import com.mapbox.services.android.navigation.ui.v5.route.NavigationMapRoute
+import com.mapbox.services.android.navigation.v5.instruction.Instruction
+import com.mapbox.services.android.navigation.v5.location.replay.ReplayRouteLocationEngine
+import com.mapbox.services.android.navigation.v5.milestone.Milestone
+import com.mapbox.services.android.navigation.v5.milestone.MilestoneEventListener
+import com.mapbox.services.android.navigation.v5.milestone.RouteMilestone
+import com.mapbox.services.android.navigation.v5.milestone.Trigger
+import com.mapbox.services.android.navigation.v5.milestone.TriggerProperty
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationOptions
+import com.mapbox.services.android.navigation.v5.navigation.NavigationEventListener
+import com.mapbox.services.android.navigation.v5.navigation.RefreshCallback
+import com.mapbox.services.android.navigation.v5.navigation.RefreshError
+import com.mapbox.services.android.navigation.v5.navigation.RouteRefresh
+import com.mapbox.services.android.navigation.v5.navigation.metrics.MapboxMetricsReporter
+import com.mapbox.services.android.navigation.v5.navigation.metrics.MetricEvent
+import com.mapbox.services.android.navigation.v5.navigation.metrics.MetricsObserver
+import com.mapbox.services.android.navigation.v5.offroute.OffRouteListener
+import com.mapbox.services.android.navigation.v5.routeprogress.ProgressChangeListener
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+import com.mapbox.turf.TurfConstants
+import com.mapbox.turf.TurfMeasurement
+import kotlinx.android.synthetic.main.activity_mock_navigation.*
+import java.lang.ref.WeakReference
+import timber.log.Timber
+
+class MockNavigationOffboardRouterActivity : AppCompatActivity(),
+    OnMapReadyCallback,
+    MapboxMap.OnMapClickListener,
+    ProgressChangeListener,
+    NavigationEventListener,
+    MilestoneEventListener,
+    OffRouteListener,
+    RefreshCallback,
+    MetricsObserver,
+    DirectionsSession.RouteObserver {
+
+    companion object {
+        private const val BEGIN_ROUTE_MILESTONE = 1001
+        private const val TWENTY_FIVE_METERS = 25.0
+    }
+
+    private var mapboxMap: MapboxMap? = null
+
+    // Navigation related variables
+    private lateinit var locationEngine: LocationEngine
+    private lateinit var navigation: MapboxNavigation
```

Comment:
```suggestion
    private val navigation: MapboxNavigation by lazy {
    MapboxNavigation(
            this,
            Utils.getMapboxAccessToken(this),
            options
        )
    }
```

Reply:
It is not needed, because
1) We initialize `navigation` instance in our own activity in `onCreate()` method.
2) To initialize it lazily, we need to pass `MapboxNavigationOptions` to it. To init `MapboxNavigationOptions` we need to pass `CustomNavigationNotification`... So it much easier to init `MapboxNavigation` as it is.